### PR TITLE
Hierarchical system bug

### PR DESCRIPTION
**Pull request by eggrobin**
Encountered while investigating #631.

### CODE REVIEW

File: physics/hierarchical_system_body.hpp [hunk 1]
```diff
@@ -58,8 +58,18 @@ HierarchicalSystem<Frame>::ToBarycentric(System& system) {
   auto const semimajor_axis_less_than = [](
       not_null<std::unique_ptr<Subsystem>> const& left,
       not_null<std::unique_ptr<Subsystem>> const& right) -> bool {
-    return left->jacobi_osculating_elements.semimajor_axis <
+    bool const semimajor_axes = left->jacobi_osculating_elements.semimajor_axis &&
```
Comment (pleroy) [inline physics/hierarchical_system_body.hpp]: Call this has_semimajor_axes (or have_).  The current identifier feels like it denotes the axis itself.  Same for mean motions.

File: physics/hierarchical_system_body.hpp [hunk 2]
```diff
@@ -58,8 +58,18 @@ HierarchicalSystem<Frame>::ToBarycentric(System& system) {
...
            right->jacobi_osculating_elements.semimajor_axis;
```
Comment (pleroy) [inline physics/hierarchical_system_body.hpp]: Indent `right` below `left`.  Same for mean_motion.

File: physics/hierarchical_system_body.hpp [hunk 3]
```diff
@@ -58,8 +58,18 @@ HierarchicalSystem<Frame>::ToBarycentric(System& system) {
...
+    bool const mean_motions = left->jacobi_osculating_elements.mean_motion &&
+           right->jacobi_osculating_elements.mean_motion;
+    CHECK(semimajor_axes || mean_motions);
```
Comment (pleroy) [inline physics/hierarchical_system_body.hpp]: Should the check be `^`?

File: physics/hierarchical_system_body.hpp [hunk 4]
```diff
@@ -58,8 +58,18 @@ HierarchicalSystem<Frame>::ToBarycentric(System& system) {
...
+    if (semimajor_axes) {
+      return left->jacobi_osculating_elements.semimajor_axis <
+             right->jacobi_osculating_elements.semimajor_axis;
+    } else {
```
Comment (pleroy) [inline physics/hierarchical_system_body.hpp]: Make this a second if, it makes the condition explicit.

File: physics/hierarchical_system_test.cpp [hunk 5]
```diff
@@ -81,5 +80,54 @@ TEST_F(HierarchicalSystemTest, HierarchicalSystem) {
...
+                 [](DegreesOfFreedom<Frame> const& dof) {
+                   return (dof.position() - Frame::origin).coordinates().x;
+                 });
+  EXPECT_THAT(x_positions,
+              ElementsAre(-1 * Metre,
+                          VanishesBefore(1 * Metre, 1),
+                          1 * Metre));
+}
+
+
```
Comment (pleroy) [inline physics/hierarchical_system_test.cpp]: Remove extra empty line.

File: physics/hierarchical_system_body.hpp (hunk 1 above)
Comment (eggrobin) [inline physics/hierarchical_system_body.hpp]: Done.

File: physics/hierarchical_system_body.hpp (hunk 2 above)
Comment (eggrobin) [inline physics/hierarchical_system_body.hpp]: Done by clang-format with the longer identifiers.

File: physics/hierarchical_system_body.hpp (hunk 3 above)
Comment (eggrobin) [inline physics/hierarchical_system_body.hpp]: No need for this. Arguably the check for `^` should be in a function in `kepler_orbit.hpp`, called by `Add`.

File: physics/hierarchical_system_body.hpp (hunk 4 above)
Comment (eggrobin) [inline physics/hierarchical_system_body.hpp]: Done.

File: physics/hierarchical_system_test.cpp (hunk 5 above)
Comment (eggrobin) [inline physics/hierarchical_system_test.cpp]: Done.

**Pull request merged**